lib/ziti_tunnel.c

```diff
@@ -174,17 +172,16 @@ static void on_dns_packet(void *arg, struct udp_pcb *pcb, struct pbuf *p,
     tunneler_context tnlr_ctx = arg;
 
     struct resolve_req *rr = calloc(1,sizeof(struct resolve_req));
-    rr->qp = p;
     rr->addr = *addr;
     rr->port = port;
     rr->tnlr_ctx = tnlr_ctx;
 
     int rc = tnlr_ctx->dns->query(tnlr_ctx->dns, p->payload, p->len, send_dns_resp, rr);
```

Comment:
this is not safe, there is no guarantee that query is a synchronous operation, `fallback_cb` is always called on a worker thread.

You need to copy payload into `resolve_req`